src/api/mod.rs

```diff
@@ -377,13 +380,23 @@ pub trait Peripheral: Send + Sync + Clone + Debug {
 }
 
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Clone)]
 pub enum CentralEvent {
     DeviceDiscovered(BDAddr),
     DeviceLost(BDAddr),
     DeviceUpdated(BDAddr),
     DeviceConnected(BDAddr),
     DeviceDisconnected(BDAddr),
+    ManufacturerAdvertisement {
+        address: BDAddr,
+        manufacturer_id: u16,
+        data: Vec<u8>,
+    },
+    ServiceAdvertisement {
+        address: BDAddr,
+        service: UUID,
+        data: Vec<u8>,
+    },
```

Comment:
Having both these events and `DeviceUpdated` seems a bit inconsistent to me: now you can get some information about devices from these specific events (as well as getting a `DeviceUpdated` event), while some you only get a `DeviceUpdated` event and have to look up the peripheral.

What would you still need to use the `DeviceUpdated` event for? Just RSSI?

Reply:
The RSSI, potentially the local name, the list of advertised services. It may also be beneficial to also include what properties have been changed in the `CentralEvent::DeviceUpdated` in for form of `bitflags`?

Reply:
Can any of these other than the RSSI actually change after a device is initially discovered?

Reply:
I see that `tx_power_level` may also change since it's part of the Assigned Number for Advertisement Data https://www.bluetooth.com/specifications/assigned-numbers/generic-access-profile/